### PR TITLE
Surface dropped inbound messages via new on_error callback

### PR DESCRIPTION
**Pull request by ryanrishi**
## Summary

`_handle_communication_created` reconciles participants before invoking the message-ready callback. When `_reconcile_participants` returns `None` (listParticipants unreachable, a 409 on participant promotion, a wrong-type agent at our address, or no resolvable customer), the handler logged a `warning` and returned early. The message-ready callback never fired and nothing was surfaced to the app — the inbound message was silently dropped and invisible.

This introduces an optional `on_error` callback on `TAC`, mirroring the existing callback pattern (`on_message_ready`/`on_interrupt`/`on_conversation_ended`):

- `TAC.on_error(callback)` registers a handler; `TAC.trigger_error(error, context)` dispatches it, supporting both sync and async handlers. Exceptions raised by the handler are logged and swallowed so error reporting never breaks the caller's flow.
- The reconcile-failure branch in `messaging.py` now logs at error level AND calls `trigger_error` with context: `conversation_id`, `channel`, and `dropped_inbound=True`.

The callback is additive and backward compatible: with no handler registered, `trigger_error` is a no-op and behavior degrades to logging.

## Type of Change

- [x] Bug fix
- [x] New feature
- [x] Documentation update

## Checklist

- [x] Tests added/updated
- [x] Documentation updated
- [ ] Tested E2E

## SDK Parity

This is the Python parity fix for [twilio/twilio-agent-connect-typescript#80](https://github.com/twilio/twilio-agent-connect-typescript/pull/80). Python required a **new** `on_error` callback because it had none — the TypeScript SDK already had `onError` on its base channel, so this brings Python to parity.

- [x] TypeScript SDK PR: twilio/twilio-agent-connect-typescript#80

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/test_error_callback.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+def get_test_config() -> dict[str, Any]:
+    return {
+        "account_sid": "ACtest123",
+        "auth_token": "test_token_123",
+        "api_key": "SK123",
+        "api_secret": "test_api_token",
+        "conversation_configuration_id": "conv_configuration_test123",
+        "phone_number": "+15551234567",
+    }
```
[the remark this answers is not in the document]
Comment (ryanrishi) [reply tests/test_error_callback.py:27]: These tests are already hermetic. tests/conftest.py defines an autouse `mock_conversation_configuration` fixture that patches `ConversationClient.get_configuration` for every test in the suite, returning a minimal configuration with `memory_store_id`. No real network call happens during `TAC(get_test_config())` init. This matches how the rest of the suite (e.g. test_tac.py, test_participant_reconciliation.py) constructs TAC, so adding per-test mocking here would duplicate the shared fixture and diverge from the repo convention.

File: README.md [hunk 2]
```diff
@@ -164,6 +164,23 @@ TAC simplifies building AI agents by handling the integration between Twilio's c
 4. **Callback Invoked**: Your `on_message_ready` callback receives user message, context, and optional memory response
 5. **Response Handling**: Your callback returns a response string that TAC routes to the appropriate channel
 
+### Callbacks
```
Comment (xinghaohuang91) [inline README.md]: I prefer to keeping readme clean. I think we will have this in the online api docs.

**Review by xinghaohuang91 — APPROVED** [review]

**Review by ryanrouleau — APPROVED** [review]
